### PR TITLE
Minor style cleanup for Module

### PR DESCRIPTION
**Pull request by steven-johnson**
All existing use was possible via the accessors and mutators, so make
the public members private. Also change list iteration to use C++11
style iteration.

(Purely stylistic and maybe overkill, but it was bothering me.)

### CODE REVIEW

File: src/Module.h [hunk 1]
```diff
@@ -47,6 +47,12 @@ class Module {
     std::string name_;
```
Comment (zvookin) [inline src/Module.h]: While we're here, the trailing underscores are not Halide style.

Comment (steven-johnson) [inline src/Module.h]: This is true, but renaming those would require renaming the accessor functions as well.

(IMHO Halide should adopt the trailing underscore style, but that's a separate issue.)

**Comment by steven-johnson** [comment]
PTAL

**Comment by abadams** [comment]
LGTM. Feel free to merge once builds go green.

**Comment by zvookin** [comment]
LGTM.

**Pull request merged**